include/motor-control/core/stepper_motor/motor_interrupt_handler.hpp

```diff
@@ -221,9 +221,36 @@ class MotorInterruptHandler {
         set_buffered_move(Move{});
     }
 
-    auto get_encoder_pulses() { return hardware.get_encoder_pulses(); }
+    auto get_encoder_pulses() {
+        if (get_encoder_overflow_status() == true) {
+            hardware.clear_encoder_SR();
+            if (enc_overflow_future_flag) {
+                enc_position_tracker = hardware.get_encoder_pulses();
+                enc_overflow_future_flag = false;
+            } else {
+                enc_position_tracker =
+                    UINT16_MAX + hardware.get_encoder_pulses();
+                enc_overflow_future_flag = true;
+            }
+        } else {
+            enc_position_tracker = hardware.get_encoder_pulses();
+        }
+        return enc_position_tracker;
+        /* This function fixes the overflow issue for each motor.
```

Comment:
kind of a nitpick but a comment like this should probably be at the start of the function, not the end